### PR TITLE
VP-6413: Add statuses to Organization, Vendor, Employee, Contact

### PR DESCRIPTION
**Pull request by Vectorfield4**
## Problem:
No status field for Organization, Vendor, Employee, Contact.

## Solution:
Add a separated (different status per entity) status field for Organization, Vendor, Employee, Contact.

## Jira:
https://virtocommerce.atlassian.net/browse/VP-6413

### Settings
VP-6413-settings
### Creating
VP-6413-creating
### Search
VP-6413-search
### Indexation
VP-6413-indexation

### CODE REVIEW

File: src/VirtoCommerce.CustomerModule.Web/Scripts/blades/customer-detail.js [hunk 1]
```diff
@@ -1,69 +1,88 @@
...
-            id: 'settingDetailChild',
-            isApiSave: true,
-            currentEntityId: 'Customer.MemberGroups',
-            parentRefresh: function (data) { $scope.groups = data; },
-            controller: 'platformWebApp.settingDictionaryController',
-            template: '$(Platform)/Scripts/app/settings/blades/setting-dictionary.tpl.html'
-        };
-        bladeNavigationService.showBlade(newBlade, blade);
-    };
+            $scope.statuseSettings = settings.get({ id: 'Customer.ContactStatuses' }, function (data) {
```
Comment (eMazeika) [inline src/VirtoCommerce.CustomerModule.Web/Scripts/blades/customer-detail.js:35]: redundant e in $scope.statusSettings

File: src/VirtoCommerce.CustomerModule.Web/Scripts/blades/customer-detail.js [hunk 2]
```diff
@@ -1,69 +1,88 @@
...
+                    blade.origEntity.status = data.defaultValue;
+                }
+            });
 
-    $scope.fetchCustomerOrganizations = function($select) {
-        $scope.fetchOrganizations($select).then(function() {
-            $scope.customerOrganizations = angular.copy($scope.organizations);
-        });
-    }
+            blade.openStatusSettingManagement = function () {
```
Comment (eMazeika) [inline src/VirtoCommerce.CustomerModule.Web/Scripts/blades/customer-detail.js:42]: Status is property of a base Member type. Move as much as possible to base memberDetailController: setting loading and openStatusSettingManagement

**Review by eMazeika — CHANGES_REQUESTED** [review]

**Review by OlegoO — DISMISSED** [review]

File: src/VirtoCommerce.CustomerModule.Web/Scripts/blades/customer-detail.js (hunk 2 above)
Comment (Vectorfield4) [reply src/VirtoCommerce.CustomerModule.Web/Scripts/blades/customer-detail.js:42]: Yep, moved.

File: src/VirtoCommerce.CustomerModule.Web/Scripts/blades/customer-detail.js (hunk 1 above)
Comment (Vectorfield4) [reply src/VirtoCommerce.CustomerModule.Web/Scripts/blades/customer-detail.js:35]: What do you mean?

**Review by OlegoO — APPROVED** [review]

**Review by eMazeika — APPROVED** [review]

**Review by OlegoO — APPROVED** [review]

**Pull request merged**